crates/apub/src/objects/post.rs

```diff
@@ -177,10 +177,14 @@ impl ApubObject for ApubPost {
       .name
       .clone()
       .or_else(|| {
-        page
-          .content
-          .clone()
-          .and_then(|c| c.replace("<p>","").replace("</p>","\n").replace("<br","\n").lines().next().map(ToString::to_string))
+        page.content.clone().and_then(|c| {
+          c.replace("<p>", "")
+            .replace("</p>", "\n")
+            .replace("<br", "\n")
```

Comment:
This is not the right way to do this, who knows what other html tags there are. The issue is that its html that's coming in, that needs to be converted to markdown.

Reply:
That is true. I already saw this example for which it would not work: https://lemmy.ml/post/741995

The best way would be to import some specialized parser. 

If no such parser is available (very unlikely), I think that replacing \</p> and \<br with newlines, and then using a regex to wipe out anything between brackets, would work for almost any case.

(I am sorry about polluting this PR with a "merge branch" commit and the revert - I was trying to merge main into my fork, and did not know that it would go into the PR.)

Reply:
That's no problem.

@Nutomic how do we convert the html -> markdown in the rest of the apub code?

Reply:
Did you already try with `page.source` like I mentioned?

Reply:
I have tried to get the content from `page.source.content`, but I am not familiar enough with Rust to understand how to do this properly, as I get an "object must have name or content" error. 

I have tested using parse_html, and it does work:

`use html2md::parse_html;
`


```
      .or_else(|| {
        page
          .content
          .clone()
          .as_ref()
          .and_then(|c| parse_html(c).lines().next().map(ToString::to_string))
      })

```

I also noticed that when loading the content it is being converted from markdown into HTML

`      content: self.body.as_ref().map(|b| markdown_to_html(b)),
`

So it should also be possible to load the content directly without that conversion step.

Reply:
Damn Mastodon doesnt send the `source` field so we have to parse the HTML anyway. You can use this command to view the Activitypub data, as you see its only HTML :(

`curl -H 'Accept: application/activity+json' https://mastodon.uno/users/informapirata/statuses/109790329058482726 | jq`

> I also noticed that when loading the content it is being converted from markdown into HTML
> 
>      content: self.body.as_ref().map(|b| markdown_to_html(b)),
>
> So maybe it is possible to load the content directly without that conversion step.

That code is executed when Lemmy sends data, its not for receiving.

Reply:
Ah, thank you! Then I think I did manage to extract the source content correctly, but the string was empty.

I have tested the parse_html approach and it does work well.

You can see the result by comparing the test titles on mander vs lemmy: 
https://mander.xyz/c/test@lemmy.ml
http://lemmy.ml/c/test

I have noticed is that titles that include Url markdown will appear to be more truncated as the markdown counts towards the char limit. Example: https://mander.xyz/post/528455

Reply:
Its also a bit odd that lemmy-ui seems to render markdown in titles now. Maybe we could also try to convert the markdown to plaintext if theres an easy way to do that. In any case this is already a clear improvement and we could merge it like this (but you need to do a bunch of cleanup).

Reply:
Would it be correct to submit a different PR that incorporates the parse_html approach instead of this one? I am not familiar with how to clean up the PR, and it now says that 10 files have been changed.

Reply:
No you can use the same PR. Easiest solution in that case might be to delete your local branch, update your main branch and then create a new branch with the same name. Then you can apply your changes and force push.

Reply:
I am still not very experienced with this. I did not use a new branch for the PR - I forked the project and edited directly to main. When I re-synced main it did deleted the commits. But that action closed the pull request automatically, and it did not re-open when I committed the new changes to main (https://github.com/Kradyz/mander-lemmy). Now I see that I should have created a new branch for making the changes.

Reply:
No worries, everyone has to learn.